src/main/java/mate/project/service/BookServiceImpl.java

```diff
@@ -31,8 +31,21 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book book = bookRepository.getBookById(id).orElseThrow(
+        Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Book with id " + id + " not found"));
         return bookMapper.toBookDto(book);
     }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto bookDto) {
```

Comment:
The `put` methods usually work like this: Update an existing object, but if it doesn't exist, create a new one. so consider changing this method to something like this:

```java
        Book modelBook = bookMapper.toModel(book);
        modelBook.setId(id);
        return bookMapper.toDto(bookRepository.save(modelBook));
```